packages/webpack-cli/lib/groups/ConfigGroup.js

```diff
@@ -117,6 +118,14 @@ class ConfigGroup extends GroupHelper {
             const newOptions = configOptions(formattedEnv, argv);
             // When config function returns a promise, resolve it, if not it's resolved by default
             newOptionsObject['options'] = await Promise.resolve(newOptions);
+        } else if (Array.isArray(configOptions) && this.args.configName) {
```

Comment:
Add a comment when this block is executed, can give example, looks good otherwise